File: 백준/Silver/13305. 주유소/주유소.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./ex.txt";
let input = fs.readFileSync(filePath).toString().trim().split("\n");
const N = input[0];
const road_len = input[1].split(" ").map(BigInt);
const oil_price = input[2].split(" ").map(BigInt);

// console.log(N, road_len, oil_price);

let cur_price = oil_price[0];
let cost = 0n;

for (let i = 0; i < N - 1; i++) {
  // 일단 가야 함.
  cost += cur_price * road_len[i];

  // 내가 더 비싸면 다음 거리는 다음 가게 가격으로 사자
  if (cur_price > oil_price[i + 1]) {
    cur_price = oil_price[i + 1];
  }
}

console.log(String(cost));

File: 백준/Silver/13305. 주유소/주유소.py
import sys

# 입력 받기
input_data = sys.stdin.read().split("\n")
N = int(input_data[0])
road_len = list(map(int, input_data[1].split()))
oil_price = list(map(int, input_data[2].split()))

# 초기화
cur_price = oil_price[0]
cost = 0

# 주유소 최적 경로 탐색
for i in range(N - 1):
    # 일단 현재 주유소에서 다음 도시까지 가야 함
    cost += cur_price * road_len[i]

    # 다음 주유소 가격이 더 싸다면, 다음 주유소에서 기름 충전
    if cur_price > oil_price[i + 1]:
        cur_price = oil_price[i + 1]

print(cost)